http-sink/main.go

```diff
@@ -39,24 +40,19 @@ func (i *arrayFlags) Set(value string) error {
 }
 
 func (hs *httpSink) sendHTTPRequest(data io.Reader) error {
-	client := &http.Client{Timeout: time.Duration(hs.timeout) * time.Second}
-	if hs.skipInsecure {
-		hs.logger.Info("Send insecure request")
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
-		client = &http.Client{Timeout: 2 * time.Second, Transport: tr}
-	}
 	req, err := http.NewRequest(hs.method, hs.url, data)
 	if err != nil {
 		return err
 	}
-	res, err := client.Do(req)
+	res, err := hs.httpClient.Do(req)
```

Comment:
Are there failure scenarios where we need to re-create the http.Client?

Reply:
the client connection will not close. it will return to the pool. we don't need to re-create it.